src/Controllers/SchemaController.js

```diff
@@ -1101,7 +1112,15 @@ export default class SchemaController {
           )} but got ${typeToString(type)}`
         );
       }
-      return undefined;
+      // If type options do not change
+      // we can safely return
+      if (isValidation || _.isEqual(expectedType, type)) {
+        return undefined;
+      } else {
```

Comment:
no need for `else` statement if returning out on line 1118

Reply:
on it